src/context.rs

```diff
@@ -53,13 +53,23 @@ impl Display for ConnectionState {
     }
 }
 
+/// Listener for application events.
+pub trait ApplicationEventListener {
+    /// Report new connection state.
+    fn connection_state_changed(&mut self, _: ConnectionState) {}
+
+    /// Report new network scanner state.
+    fn network_scanner_state_changed(&mut self, _: bool) {}
```

Comment:
Adding another trait just for providing default method implementations is not very common pattern in Rust and IMHO it'd be an unnecessary bloat. On the other hand, providing default method implementations in traits is very common. Regarding the arguments, I replaced their names with `_` because of Clippy. I could have used `_state` and `_scanning` but I don't like identifiers starting with underscore :) .

Anyway, this issue can be easily resolved by improving the doc string for the methods.

Reply:
Improving the doc string would be nice.

But still imagine a situation when the function has more than one parameter.
```rust
fn other_function(&self, _: usize, _: usize);
```
In the doc string you would need to write something like `the first usize argument`, `the second usize argument`. Which seems harder to process for the reader.